lib/plugins/blocks.js

```diff
@@ -147,6 +147,41 @@ function inject(bot) {
     }
   }
 
+  function blockIsNotEmpty(pos) {
+    var block = bot.blockAt(pos);
+    return block!==null && block.boundingBox!=="empty";
+  }
+
+  // maybe this should be moved to math.js instead?
+  function visiblePosition(a,b) {
+    var v=b.minus(a);
+    var t=Math.sqrt(v.x*v.x+v.y*v.y+v.z*v.z);
+    v=v.scaled(1/t);
+    v=v.scaled(1/5);
+    var u=t*5;
+    var na;
+    for(var i=1;i<u;i++) {
+      na=a.plus(v);
+      // check that blocks don't inhabit the same position
+      if(!na.floored().equals(a.floored())) {
+        // check block is not transparent
+        if(blockIsNotEmpty(na)) return false;
+      }
+      a=na;
+    }
+    return true;
+  }
+
+  // if passed in block is within line of sight to the bot, returns true
+  // also works on anything with a position value
+  function canSeeBlock(block) {
+    // this emits a ray from the center of the bots body to the block
+    if(visiblePosition(bot.entity.position.offset(0,bot.entity.height*0.5,0),block.position) ) {
```

Comment:
Isn't the height, if the bot player is standing or otherwise upright, relative to the Z (third) dimension? If so, shouldn't this be:

```
... bot.entity.position.offset(0,0,bot.entity.height*0.5) ...
```

Also, why half the bot entity's height? Why not the full height or 80% (roughly where the eyes would be)?